refresher/git_tutorial.md

```diff
@@ -0,0 +1,241 @@
+- Understand the basics of how automated version control systems work.
+
+### Tutorial steps
+
+Version control systems are tools that keep track of changes to documents. There are
+many benefits to using version control systems, including:
+
+- Having a complete change history for all files.
+- Collaborating effectively through branching and merging.
+- Relating changes to project management goals.
+
+Version control systems are most commonly used for code, but are useful for many
+other tasks such as writing papers, proposals, and presentations.
+
+## Setting up git
+
+### Objectives
+
+- Configure `git` the first time it is used on a computer.
+- Understand the meaning of the `--global` configuration flag.
+
+### Tutorial steps
+
+In this section, we will configure some Git settings.
+
+1. Open terminal (macOS) or Git Bash (windows).
+2. Set your name:
+
+    `git config --global user.name "Your name"`
+
+3. Set your email (see note below for instructions to keep your email private):
+
+    `git config --global user.email "Your email"`
+
+4. Set your preferred line endings:
+
+    `git config --global core.autocrlf input` (macOS)
+
+    `git config --global core.autocrlf true` (windows)
+
+5. Set your preferred text editor (See
+  [git-scm Appendix C](https://www.git-scm.com/book/en/v2/Appendix-C%3A-Git-Commands-Setup-and-Config)
+  for options):
+
+   > **NOTE**: The line below needs to be adjusted for your particular text editor using the
+   > options in the link provided above. For example, the command for setting Visual Studio
+   > Code as your preferred text editor is `git config --global core.editor "code --wait"`.
+
+    `git config --global core.editor "editor name"`
+
+6. Set a default branch name:
+
+    `git config --global init.defaultBranch main`
+
+7. Check your settings:
+
+    `git config --list`
+
+    `git config user.name`
+
+> **NOTE**: If you wish to use a private email with GitHub, then you can use the format
+> `git config --global user.email username@users.noreply.github.com`, where `username` is
+> replaced by your GitHub username.
+
+Many other settings are [configurable in Git](https://www.git-scm.com/book/en/v2/Customizing-Git-Git-Configuration).
+You can also get help using `git config --help`.
+
+## Creating a repository
+
+### Objectives
+
+- Create a local Git repository.
+- Describe the purpose of the `.git` directory.
+
+### Tutorial steps
+
+In this section, we will create a repository.
+
+1. Create a directory in the `Desktop` folder:
+
+    ```
+    cd ~/Desktop
+    mkdir learn-git
+    cd learn-git
+    ```
+
+2. Tell Git to make `learn-git` a repository:
+
+    `git init`
+
+3. View the new `.git` hidden directory created by Git:
+
+    `ls -a`
+
+4. Check the status of the Git repository:
+
+    `git status`
+
+## Tracking changes
+
+### Objectives
+
+- Complete the modify-add-commit cycle.
+- Understand how to write descriptive commit messages.
```

Comment:
Might help to have a figure somewhere here, I'm thinking of this one from https://swcarpentry.github.io/git-novice/04-changes/index.html#staging-area:

![image](https://user-images.githubusercontent.com/23487320/129636790-fc97c0d8-6e63-4fd5-9f32-75b4385247db.png)


```suggestion
## Tracking changes

### Objectives

- Complete the modify-add-commit cycle.
- Understand how to write descriptive commit messages.

![git staging area](https://raw.githubusercontent.com/swcarpentry/git-novice/v2019.06.1/fig/git-staging-area.svg)
```